extension/src/counter_agg/gauge.rs

```diff
@@ -0,0 +1,579 @@
+    }
+    summary_buffer.push(new_summary.build());
+}
+
+#[pg_extern(immutable, parallel_safe, strict, schema = "toolkit_experimental")]
+pub fn gauge_summary_trans_serialize(state: Internal) -> bytea {
+    let state: &mut CounterSummaryTransState = unsafe { state.get_mut().unwrap() };
+    state.combine_summaries();
+    crate::do_serialize!(state)
+}
+
+#[pg_extern(strict, immutable, parallel_safe, schema = "toolkit_experimental")]
+pub fn gauge_summary_trans_deserialize(bytes: bytea, _internal: Internal) -> Internal {
+    gauge_summary_trans_deserialize_inner(bytes).internal()
+}
+fn gauge_summary_trans_deserialize_inner(bytes: bytea) -> Inner<CounterSummaryTransState> {
+    let c: CounterSummaryTransState = crate::do_deserialize!(bytes, CounterSummaryTransState);
+    c.into()
+}
+
+#[pg_extern(immutable, parallel_safe, schema = "toolkit_experimental")]
+pub fn gauge_agg_trans(
+    state: Internal,
+    ts: Option<crate::raw::TimestampTz>,
+    val: Option<f64>,
+    bounds: Option<tstzrange>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Internal {
+    gauge_agg_trans_inner(unsafe { state.to_inner() }, ts, val, bounds, fcinfo).internal()
+}
+fn gauge_agg_trans_inner(
+    state: Option<Inner<CounterSummaryTransState>>,
+    ts: Option<crate::raw::TimestampTz>,
+    val: Option<f64>,
+    bounds: Option<tstzrange>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Inner<CounterSummaryTransState>> {
+    unsafe {
+        in_aggregate_context(fcinfo, || {
+            let p = match (ts, val) {
+                (_, None) => return state,
+                (None, _) => return state,
+                (Some(ts), Some(val)) => TSPoint { ts: ts.into(), val },
+            };
+            match state {
+                None => {
+                    let mut s = CounterSummaryTransState::new();
+                    if let Some(r) = bounds {
+                        s.bounds = get_range(r.0 as *mut pg_sys::varlena);
+                    }
+                    s.push_point(p);
+                    Some(s.into())
+                }
+                Some(mut s) => {
+                    s.push_point(p);
+                    Some(s)
+                }
+            }
+        })
+    }
+}
+
+#[pg_extern(immutable, parallel_safe, schema = "toolkit_experimental")]
+pub fn gauge_agg_trans_no_bounds(
+    state: Internal,
+    ts: Option<crate::raw::TimestampTz>,
+    val: Option<f64>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Internal {
+    gauge_agg_trans_inner(unsafe { state.to_inner() }, ts, val, None, fcinfo).internal()
+}
+
+#[pg_extern(immutable, parallel_safe, schema = "toolkit_experimental")]
+pub fn gauge_agg_summary_trans(
+    state: Internal,
+    value: Option<CounterSummary>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Internal {
+    gauge_agg_summary_trans_inner(unsafe { state.to_inner() }, value, fcinfo).internal()
+}
+fn gauge_agg_summary_trans_inner(
+    state: Option<Inner<CounterSummaryTransState>>,
```

Comment:
This feels weird still where it's a CounterSummary...

Reply:
Is this about the name or something else? Other than the name, I think this is the right structure. Happy to discuss more.

Reply:
Made the changes we discussed.